### ISSUE TITLE
Training 2D pose model with combined datasets COCO, AI, OccHuman

### ISSUE THREAD

**Issue by khanhha** [open-mmlab/mmpose#662]
Hi MMPose Team,

I would like to ask a question about training a 2D pose model with combined datasets COCO, AI, OccHuman, etc.
There would be a much higher amount of training data by combining these datasets, so I think it would be interesting to discuss various approaches to the problem.  In the below, I summarize two main approaches based on my reserach.

1. One-stage training by disabling training supervision on unlabeled key-points: First convert all datasets to the same format, and mark missing key-points as unlabeled. For example, AI has 14 key-points while COCO has 17 key-points. The combined dataset will have 17 key-points, but the 3 missing key-points from AI are set to unlabeled. During the training, the supervision on unlabeled key-poitns will be disabled,

2. Two-stage training:  find a shared/minimum keypoint format from all datasets and perform an initial training on this shared dataset. In the second stage, change the number of output channels and then fine-tune on a target dataset. For example, for combining COCO and AI, we can first train a model with 14 keypoints format of the shared dataset. In the second step, we fine-tune the pre-trained model from the first step but with 17 output channels on the COCO dataset.

The second approach is less convenient because it requires multi-stages. But it seems to work, as tested in [this research](https://arxiv.org/pdf/2002.00537.pdf). The first approach is more convenient but I am not sure if it works or not.

Could you please let me know your opinions about the first approach? or other solutions for joint training that you can think of?

I would really appreciate it.

Best
Khanh Ha

**Comment by jin-s13** [comment]
In my experience, finetuning the model on the target dataset (after jointly training models on multiple datasets), always improves the performance. However, using different joint-training schemes will produce similar results. It does not matter whether you use the intersection (the mentioned first approach) and/or union (the mentioned second approach) of different sets of keypoints.

**Comment by jin-s13** [comment]
I think it is ok to skip some supervision, i.e. setting zero-losses for these unlabeled keypoints. 
In fact, we also intentionally skip some keypoints in the ohkm(online hard keypoint mining) loss.

**Comment by khanhha** [comment]
@jin-s13 
thank you very much for your response
I will update the experiment result of the first approach here after it's done. Hopefully, it can give benefit to others.
Best
